pipeline/util/urllib.py

```diff
@@ -4,25 +4,41 @@
 import tempfile
 from datetime import datetime
 from email.utils import parsedate_to_datetime
-from typing import Union
+from typing import Optional, Union
 
 import requests
+from requests.auth import HTTPBasicAuth
 
 user_agent = 'IPL (MobiData-BW) +https://github.com/mobidata-bw/ipl-dagster-pipeline'
 
 
-def get(url: str, timeout: int, headers: Union[dict[str, str], None] = None, stream: bool = False):
+def get(
+    url: str,
+    timeout: int,
+    headers: Union[dict[str, str], None] = None,
+    stream: bool = False,
+    basic_auth_user: Optional[str] = None,
+    basic_auth_password: Optional[str] = None,
+):
     if headers is None:
         headers = {}
     headers['User-Agent'] = user_agent
-    return requests.get(url, headers=headers, timeout=timeout, stream=stream)
+
+    if basic_auth_user is not None and basic_auth_password is not None:
```

Comment:
If you want you can simplify that as `auth: tuple[str, str] = (user, password)` which is a shortcut to `HTTPBasicAuth`.